pkg/storage/streams/store.go

```diff
@@ -413,3 +424,11 @@ func (lr *lazySegmentRanger) Range(ctx context.Context, offset, length int64) (i
 	}
 	return lr.ranger.Range(ctx, offset, length)
 }
+
+// CancelHandler handles clean up of segments on receiving CTRL+C
+func (s *streamStore) CancelHandler(ctx context.Context, totalSegments int64, path paths.Path) {
+	for i := 0; i < int(totalSegments); i++ {
+		currentPath := fmt.Sprintf("s%d", i)
+		_ = s.segments.Delete(ctx, path.Prepend(currentPath))
```

Comment:
How about logging the error?

Reply:
the reason I didn't log the err is, this function will loop and  do as much cleaning  as possible, irrespective of error segment by segment. If for any reason one segment cannot be deleted, and errors out, my thought processes was to ignore and go ahead and clean next segment. Basically reclaim as much space as we can. If I add error and return when not nil, it stops the further purging. Those are my thoughts... please let me know how you want me to handle err?

Reply:
I have nothing against what you are saying. Just if deleting a segment fails, let's log the error, so we know it, and continue with an attempt to delete the next segment.

Reply:
adding the error log..